src/Components/test/testassets/Components.TestServer/RazorComponents/Root.razor

```diff
@@ -0,0 +1,43 @@
+﻿@using Components.TestServer.RazorComponents.Pages.Forms
+@using Microsoft.AspNetCore.Components.Web
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" />
+    <base href="/subdir/" />
+    <HeadOutlet />
+</head>
+<body>
+    <ServerApp @rendermode="@RenderMode.Server" />
+
+    <script src="_framework/blazor.web.js" autostart="false"></script>
+    <script>
+        const suppressEnhancedNavigation = sessionStorage.getItem('suppress-enhanced-navigation') === 'true';
+        sessionStorage.removeItem('suppress-enhanced-navigation');
+
+        function callBlazorStart() {
+            Blazor.start({
+                ssr: {
+                    disableDomPreservation: suppressEnhancedNavigation,
+                },
+                webAssembly: {
+                    loadBootResource: function (type, name, defaultUri, integrity) {
+                        return `WasmMinimal/_framework/${name}`;
+                    },
+                },
+            });
+        }
```

Comment:
Do we definitely have to duplicate all this? It does make it quite difficult to maintain if we have many copies and don't even track which ones are applicable to which test.

Reply:
Other than the suppress enhanced navigation, I believe we do. We can put it in a separate component if we want to.

The reason I chose to create a separate startup and set of tests is because I wanted to validate that we were able to prerender, reuse the results from asp.net core routing and then start the app in interactive mode (so that the router did the parsing) and made sure that worked, akin to `Blazor Server` but with the new APIs.